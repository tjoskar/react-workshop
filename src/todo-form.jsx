import * as React from 'react';

export class TodoForm extends React.Component {
    
  state = { text: '' };

  handleChange = e => {
    this.setState({ text: e.target.value });
  }

  handleSubmit = e => {
    e.preventDefault();
    this.props.newItem(this.state.text);
    this.setState({ text: '' });
  }

  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <input onChange={this.handleChange} value={this.state.text} />
        <button>Add item</button>
      </form>
    );
  }

}
