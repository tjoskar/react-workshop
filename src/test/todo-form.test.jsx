import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { spy } from 'simple-spy';
import { TodoForm } from '../todo-form';

test('Render form', () => {
  const tree = shallow(
    <TodoForm />
  );

  expect(toJson(tree)).toMatchSnapshot();
});

test('Return the state value on save', () => {
  // Arrange
  const newItem = spy(() => null);
  const tree = shallow(
    <TodoForm newItem={newItem} />
  );
  tree.setState({ text: 'Winter is coming' });
  const submitEvent = {
    preventDefault: spy(() => null)
  }

  // Act
  tree.find('form').simulate('submit', submitEvent);

  // Assert
  expect(newItem.callCount).toBe(1);
  expect(newItem.args[0][0]).toBe('Winter is coming');
  expect(submitEvent.preventDefault.callCount).toBe(1);
});
