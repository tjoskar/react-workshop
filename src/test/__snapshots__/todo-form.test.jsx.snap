// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render form 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    value=""
  />
  <button>
    Add item
  </button>
</form>
`;
